src/main/java/reposense/authorship/model/AuthorshipSummary.java

```diff
@@ -15,40 +16,71 @@
     private final Map<Author, LinkedHashMap<String, Integer>> authorFileFormatContributionMap;
+    private final Map<Author, LinkedHashMap<String, Integer>> authorGroupContributionMap;
 
-    public AuthorshipSummary(List<FileResult> fileResults, List<Author> authors, List<Format> formats) {
+    public AuthorshipSummary(List<FileResult> fileResults, List<Author> authors, List<Format> formats,
+        List<Group> groups) {
         this.fileResults = fileResults;
         authorFinalContributionMap = new HashMap<>();
         authorFileFormatContributionMap = new HashMap<>();
+        authorGroupContributionMap = new HashMap<>();
 
         // initialise each author contribution to be 0
         authors.forEach((author) -> authorFinalContributionMap.put(author, 0));
+
+        // file format contribution
         authors.forEach((author) -> {
             LinkedHashMap<String, Integer> defaultFileFormatContribution = new LinkedHashMap<>();
             for (Format format : formats) {
                 defaultFileFormatContribution.put(format.toString(), 0);
             }
             authorFileFormatContributionMap.put(author, defaultFileFormatContribution);
         });
+
+        // group contribution
+        if (!groups.isEmpty()) {
+            authors.forEach((author) -> {
+                LinkedHashMap<String, Integer> defaultGroupContribution = new LinkedHashMap<>();
+                for (Group group : groups) {
+                    defaultGroupContribution.put(group.toString(), 0);
+                }
+                defaultGroupContribution.put(Group.DEFAULT_GROUP, 0);
+                authorGroupContributionMap.put(author, defaultGroupContribution);
+            });
+        } else {
+            authors.forEach((author) -> authorGroupContributionMap.put(author, new LinkedHashMap<>()));
+        }
     }
 
     /**
      * Increments the contribution count of {@code author} and the corresponding file format specified by
      * {@code filePath} by one.
      */
-    public void addAuthorContributionCount(Author author, String filePath) {
+
+    public void addAuthorContributionCount(Author author, String filePath, String group) {
         authorFinalContributionMap.put(author, authorFinalContributionMap.get(author) + 1);
 
+        // Add file format contribution count
         Map<String, Integer> fileFormatContributionMap = authorFileFormatContributionMap.get(author);
         String fileFormat = filePath.substring(filePath.lastIndexOf('.') + 1);
         fileFormat = fileFormat.isEmpty() ? "others" : fileFormat;
         fileFormatContributionMap.put(fileFormat, fileFormatContributionMap.getOrDefault(fileFormat, 0) + 1);
+
+        // Add group contribution count
+        if (group != null) {
+            Map<String, Integer> groupContributionMap = authorGroupContributionMap.get(author);
+            groupContributionMap.put(group, groupContributionMap.getOrDefault(group, 0) + 1);
```

Comment:
It seems like on the UI, you only utilizes one of them; if the group exists, you don't show the formats. 
In that case, why do you still need to calculate formats for the repo if you're not utilizing them?
As a benefit, you wouldn't even need to modify the frontend that much if that is the case.

Reply:
File format information is still needed for the checkboxes in the Code Panel when the `breakdown by file format/group` option is not selected, as the default view does not make use of the breakdown by grouping.

Reply:
Wouldn't it be better to retain the file format information for such repos as well, in case the user specifies it in `repo-config.csv`? @damithc your opinion?

Reply:
@eugenepeh there are 2 maps being passed to front-end now: `authorFileFormatContributionMap` and `authorGroupContributionMap`. If we are merging it into 1 map to be passed, what should it be renamed to? `authorBreakdownContributionMap`? References in front-end will also have to be changed accordingly e.g. `isSelectedFileTypes` and `isSelectedGroups` to `isSelectedBreakdown`

Reply:
If let's say they were to be mutually exclusive, we can just simplify the 2 into a single element such that there is no distinction between `java` and `code` on the frontend. This helps to enhance maintainability and computation. 

Do consult @damithc 's opinion on the ideal outcome though.

Reply:
Do what is optimal for the current feature set. Let's not complicate the code for the sake of possible future features.

Reply:
> If let's say they were to be mutually exclusive, we can just simplify the 2 into a single element such that there is no distinction between `java` and `code` on the frontend. This helps to enhance maintainability and computation.
> 
> Do consult @damithc 's opinion on the ideal outcome though.

@eugenepeh I've updated the implementation accordingly, to use `format` to refer to both `fileFormat` and `group`. So now, there is no distinction between `java` and `code`.